deletiontests/azure/azure_delete_test.go

```diff
@@ -0,0 +1,188 @@
+package ingress
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/apimachinery/pkg/api/errors"
+	capzV1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	azureclient "github.com/giantswarm/sonobuoy-plugin/v5/pkg/azure/client"
+	"github.com/giantswarm/sonobuoy-plugin/v5/pkg/azure/credentials"
+	"github.com/giantswarm/sonobuoy-plugin/v5/pkg/ctrlclient"
+)
+
+func Test_AzureDelete(t *testing.T) {
+	var err error
+
+	ctx := context.Background()
+
+	logger, err := micrologger.New(micrologger.Config{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	provider, exists := os.LookupEnv("PROVIDER")
+	if !exists {
+		t.Fatal("missing PROVIDER environment variable")
+	}
+
+	if provider != "azure" {
+		logger.Debugf(ctx, "Only Azure provider is supported by this test, skipping")
+		return
+	}
```

Comment:
I think this is fine for now and we already discussed it last week, but with my experience from multi-AZ test, it's not a big change to make this test provider independent. See https://github.com/giantswarm/sonobuoy-plugin/blob/add-azs-test/tests/availabilityzones/spec.go#L11-L15 & https://github.com/giantswarm/sonobuoy-plugin/blob/add-azs-test/tests/availabilityzones/provider.go#L40-L49

I'll refactor that stuff into `pkg` once I get there so it becomes easier to use it from all tests.

Reply:
ok thanks